youtube2zim/scraper.py

```diff
@@ -585,53 +588,102 @@ def upload_to_cache(self, key, video_path, encoder_version):
         return True
 
     def download_video_files_batch(self, options, videos_ids):
+
         succeeded = []
         failed = []
 
+        def process_thumbnail(thumbnail_path, preset_thumbnail):
+            # thumbnail might be WebP as .webp, JPEG as .jpg or WebP as .jpg
+            tmp_thumbnail = thumbnail_path
+            if not thumbnail_path.exists():
+                logger.debug("We don't have video.webp, thumbnail is .jpg")
+                tmp_thumbnail = thumbnail_path.with_suffix(".jpg")
+
+            # resize thumbnail. we use max width:248x187px in listing
+            # but our posters are 480x270px
+            resize_image(
+                tmp_thumbnail,
+                width=480,
+                height=270,
+                method="cover",
+                allow_upscaling=True,
+            )
+            optimize_image(tmp_thumbnail, thumbnail_path, **preset_thumbnail.options)
+            return True
+
+        def record_success_and_log(video_id):
+            succeeded.append(video_id)
+            logger.debug(
+                f"Successfully downloaded and processed assets for video ID - {video_id}"
+            )
+
         # set preset
-        preset = {"mp4": VideoMp4Low}.get(self.video_format, VideoWebmLow)()
+        preset_video = {"mp4": VideoMp4Low}.get(self.video_format, VideoWebmLow)()
+        preset_thumbnail = WebpMedium()
 
         for video_id in videos_ids:
             options_copy = options.copy()
             video_location = options_copy["y2z_videos_dir"].joinpath(video_id)
-            downloaded_from_cache = False
+            video_downloaded_from_cache = False
+            thumbnail_downloaded_from_cache = False
             if self.s3_storage:
-                s3_key = f"{self.video_format}/{self.video_quality}/{video_id}"
+                s3_key_video = f"{self.video_format}/{self.video_quality}/{video_id}"
+                s3_key_thumbnail = f"thumbnails/{video_id}"
                 video_path = video_location.joinpath(f"video.{self.video_format}")
+                thumbnail_path = video_location.joinpath("video.webp")
                 logger.debug(f"Attempting to download {video_id} from cache...")
-                downloaded_from_cache = self.download_from_cache(
-                    s3_key, video_path, preset.VERSION
+                video_downloaded_from_cache = self.download_from_cache(
+                    s3_key_video, video_path, preset_video.VERSION
+                )
+                thumbnail_downloaded_from_cache = self.download_from_cache(
```

Comment:
this method has become too large and hard to read with many conditions.

Because neither S3 DL/UL nor youtube-dl save requests for getting the video and the thumbnail together, I suggest we split this into distinct (similar) functions for the video and the thumbnail. It should make it all a lot clearer. Might remove the need for that `record_success_and_log` that I don't like either

Reply:
Done. Exploded into two methods